02_C# Fundamentals/SystemWatcherApp/SystemWatcherApp/Program.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using SystemWatcherApp.Configs;
+using SystemWatcherApp.Resources;
+
+namespace SystemWatcherApp
+{
+    class Program
+    {
+        readonly static List<string> startDirectories;
+        readonly static List<FileSystemWatcher> watchers;
+        readonly static string defaultDirectory = @"D:\_02_BCL\Default\";
+        static CultureInfo currentCulture;
+        static List<RuleElement> rules;
+
+        static Program()
+        {
+            rules = new List<RuleElement>();
+            startDirectories = new List<string>();
+            watchers = new List<FileSystemWatcher>();
+        }
+
+        static void Main(string[] args)
+        {
+            Console.OutputEncoding = Encoding.UTF8;
+
+            WatcherSectionGroup watcherSectionGroup = ConfigurationManager.OpenExeConfiguration(ConfigurationUserLevel.None).SectionGroups["watcherSection"] as WatcherSectionGroup;
+
+            foreach (ConfigurationSection section in watcherSectionGroup.Sections)
+            {
+                if (section.GetType() == typeof(RuleSection))
+                {
+                    RuleSection ruleSection = (RuleSection)section;
+                    RuleElementCollection ruleElementCollection = ruleSection.Rules;
+                    foreach (RuleElement ruleElement in ruleElementCollection)
+                    {
+                        rules.Add(ruleElement);
+                    }
+                }
+
+                else if (section.GetType() == typeof(LocaleSection))
+                {
+                    LocaleSection localeSection = (LocaleSection)section;
+                    switch (localeSection.LocaleCode)
+                    {
+                        case "ru":
+                            currentCulture = new CultureInfo("ru-RU");
+                            break;
+                        default:
+                            currentCulture = new CultureInfo("en-US");
+                            break;
+                    }
+                }
+
+                else if (section.GetType() == typeof(TrackingFoldersSection))
+                {
+                    TrackingFoldersSection trackingFoldersSection = (TrackingFoldersSection)section;
+                    TrackingFolderElementCollection trackingFolderElementCollection = trackingFoldersSection.TrackingFolders;
+                    foreach (TrackingFolderElement trackingFolderElement in trackingFolderElementCollection)
+                    {
+                        startDirectories.Add(trackingFolderElement.Address);
+                    }
+                }
+            }
+
+            Thread.CurrentThread.CurrentUICulture = currentCulture;
+
+            Run();
+        }
+
+        public static void Run()
```

Comment:
We can clearly see different things happening in this method: creating folders, creating file system watcher, running the endless cycle. It is better the keep such things in a separate methods, each of which has it's own specific purpose and is short and easy to understand. You partly did this by separating the method `Run()` from the rest of what is happening in `Main()`, but `Run` is too generic, it doesn't mean anything specific and it still does a number of things not related to each other.

Reply:
I have restructured that